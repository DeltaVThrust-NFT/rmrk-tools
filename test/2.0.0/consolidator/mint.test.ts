import { Consolidator } from "../../../src/rmrk2.0.0";
import {
  createCollectionMock,
  getBlockCallsMock,
  getBobKey,
  getRemarksFromBlocksMock,
  mintNftMock,
  mintNftMock2,
} from "../mocks";
import { cryptoWaitReady } from "@polkadot/util-crypto";

beforeAll(async () => {
  return await cryptoWaitReady();
});

describe("rmrk2.0.0 Consolidator: MINT", () => {
  const getSetupRemarks = () => [
    ...getBlockCallsMock(createCollectionMock().create()),
    ...getBlockCallsMock(mintNftMock().mint()),
    ...getBlockCallsMock(mintNftMock2().mint()),
  ];

  it("Should mint a NFT and make another account owner", async () => {
    const remarks = getRemarksFromBlocksMock([
      ...getBlockCallsMock(createCollectionMock().create()),
      ...getBlockCallsMock(mintNftMock().mint(getBobKey().address)),
    ]);
    const consolidator = new Consolidator();
    expect(await consolidator.consolidate(remarks)).toMatchSnapshot();
  });

  it("Should mint a NFT and make another NFT an owner", async () => {
    const remarks = getRemarksFromBlocksMock([
      ...getBlockCallsMock(createCollectionMock().create()),
      ...getBlockCallsMock(mintNftMock2().mint()),
      ...getBlockCallsMock(mintNftMock().mint(mintNftMock2(4).getId())),
    ]);
    const consolidator = new Consolidator();
    expect(await consolidator.consolidate(remarks)).toMatchSnapshot();
  });

  it("Should not allow to mint a NFT without a calss", async () => {
    const remarks = getRemarksFromBlocksMock([
      ...getBlockCallsMock(mintNftMock().mint()),
    ]);
    const consolidator = new Consolidator();
    expect(await consolidator.consolidate(remarks)).toMatchSnapshot();
  });
});
