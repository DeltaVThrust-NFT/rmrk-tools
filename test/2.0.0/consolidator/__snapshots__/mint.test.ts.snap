// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rmrk2.0.0 Consolidator: MINT Should mint a NFT and make another NFT an owner 1`] = `
Object {
  "bases": Array [],
  "collections": Array [
    Collection {
      "block": 2,
      "changes": Array [],
      "id": "d43593c715a56da27d-KANARIABIRDS",
      "issuer": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
      "max": 0,
      "metadata": "https://some.url",
      "symbol": "KANARIABIRDS",
    },
  ],
  "invalid": Array [],
  "nfts": Array [
    Object {
      "block": 3,
      "burned": "",
      "changes": Array [],
      "children": Array [],
      "collection": "d43593c715a56da27d-KANARIABIRDS",
      "forsale": 0n,
      "id": "3-d43593c715a56da27d-KANARIABIRDS-KANR-0000000000000888",
      "metadata": undefined,
      "owner": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
      "priority": Array [],
      "reactions": Object {},
      "resources": Array [],
      "rootowner": "",
      "sn": "0000000000000888",
      "symbol": "KANR",
      "transferable": 1,
    },
    Object {
      "block": 4,
      "burned": "",
      "changes": Array [],
      "children": Array [],
      "collection": "d43593c715a56da27d-KANARIABIRDS",
      "forsale": 0n,
      "id": "4-d43593c715a56da27d-KANARIABIRDS-KANR-0000000000000777",
      "metadata": undefined,
      "owner": "4-d43593c715a56da27d-KANARIABIRDS-KANR-0000000000000888",
      "priority": Array [],
      "reactions": Object {},
      "resources": Array [],
      "rootowner": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
      "sn": "0000000000000777",
      "symbol": "KANR",
      "transferable": 1,
    },
  ],
}
`;

exports[`rmrk2.0.0 Consolidator: MINT Should mint a NFT and make another account owner 1`] = `
Object {
  "bases": Array [],
  "collections": Array [
    Collection {
      "block": 2,
      "changes": Array [],
      "id": "d43593c715a56da27d-KANARIABIRDS",
      "issuer": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
      "max": 0,
      "metadata": "https://some.url",
      "symbol": "KANARIABIRDS",
    },
  ],
  "invalid": Array [],
  "nfts": Array [
    Object {
      "block": 3,
      "burned": "",
      "changes": Array [],
      "children": Array [],
      "collection": "d43593c715a56da27d-KANARIABIRDS",
      "forsale": 0n,
      "id": "3-d43593c715a56da27d-KANARIABIRDS-KANR-0000000000000777",
      "metadata": undefined,
      "owner": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
      "priority": Array [],
      "reactions": Object {},
      "resources": Array [],
      "rootowner": "",
      "sn": "0000000000000777",
      "symbol": "KANR",
      "transferable": 1,
    },
  ],
}
`;

exports[`rmrk2.0.0 Consolidator: MINT Should not allow to mint a NFT without a calss 1`] = `
Object {
  "bases": Array [],
  "collections": Array [],
  "invalid": Array [
    Object {
      "block": 2,
      "caller": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
      "message": "NFT referencing non-existant parent collection d43593c715a56da27d-KANARIABIRDS",
      "object_id": "2-d43593c715a56da27d-KANARIABIRDS-KANR-0000000000000777",
      "op_type": "MINT",
    },
  ],
  "nfts": Array [],
}
`;
